dkg/nodesigs.go

```diff
@@ -85,21 +129,53 @@ func (n *nodeSigBcast) exchange(
 	}
 
 	ret := make([][]byte, n.operatorAmt)
-	ret[nodeIdx.PeerIdx] = sig
+	ret[n.nodeIdx.PeerIdx] = sig
+
+	for {
```

Comment:
this will spin using 100% CPU, suggest using a ticker with period 100ms to trigger checking the map.